sdks/java/io/fileschematransform/build.gradle

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins { id 'org.apache.beam.module' }
+applyJavaNature(
+        automaticModuleName: 'org.apache.beam.sdk.io.fileschematransform'
+)
+
+description = "Apache Beam :: SDKs :: Java :: IO :: File Schema Transform"
+ext.summary = "Defines transforms for File reading and writing support with Schema Transform"
+
+// Don't pull in newer versions of the checker framework from dependencies.
+// TODO(BEAM-11125) Remove this when Beam upgrades to newest checker framework version.
+configurations.implementation {
+    resolutionStrategy {
+        force library.java.checker_qual
+    }
+}
+
+// exclude auto-generated classes and integration tests
+def jacocoExcludes = [
+        '**/AutoValue_*',
+        '**/*IT*',
+]
+
+dependencies {
+    implementation project(path: ":sdks:java:core", configuration: "shadow")
+    testImplementation library.java.junit
+    testImplementation project(path: ":sdks:java:core", configuration: "shadow")
```

Comment:
The dependencies were kept purposely minimal and will be added incrementally as work on #24464 progresses.